internal/scorecard/plugins/olm_tests.go

```diff
@@ -433,32 +412,53 @@ func (t *SpecDescriptorsTest) Run(ctx context.Context) *schelpers.TestResult {
 		res.State = scapiv1alpha1.ErrorState
 		return res
 	}
-	if t.CR.Object["spec"] == nil {
+
+	return checkCRConfiguration(t.CR, t.CSV, specDescriptor, res)
+}
+
+func checkCRConfiguration(cr *unstructured.Unstructured, csv *olmapiv1alpha1.ClusterServiceVersion, descriptor string, res *schelpers.TestResult) *schelpers.TestResult {
```

Comment:
```suggestion
func addStatusDescriptor(cr *unstructured.Unstructured, csv *olmapiv1alpha1.ClusterServiceVersion, descriptor string, res *schelpers.TestResult) *schelpers.TestResult {
```
WDYT?

Reply:
I agree that the name might not be the best. 

As per my understanding the code checks the Kubernetes `spec` or `status` object against the `CRD`.

So a better name could be `calculateCRScore`?

Reply:
It is actually is checking the spec and status descriptors in the CSV regard the CRD's definitions. Then, if all be filed it will ++ points in order to allow it pass in the scorecard checks.  
See [here](https://github.com/dev4devs-com/postgresql-operator/blob/master/deploy/olm-catalog/postgresql-operator/0.1.1/postgresql-operator.v0.1.1.clusterserviceversion.yaml#L73) and example of what is checked here. 

So, wdyt about something such as `[check/verify]OwnedCSVDescriptors`?

Reply:
I updated the suggestion with `checkOwnedCSVDescriptors `.

Reply:
Aha, I see. I'm fairly new to operators for K8s and operator-sdk. So this is good, I learn alot :)

I agree with your suggestion and with update accordingly. 👍